nautobot/extras/management/__init__.py

```diff
@@ -56,8 +93,8 @@ def populate_status_choices(apps, schema_editor, **kwargs):
 
     When it is ran again post-migrate will be a noop.
     """
-
     app_config = apps.get_app_config("extras")
+    # TODO: why can't/shouldn't we pass `apps` through to create_custom_statuses? We get failures if we do, but why?
```

Comment:
Because `apps` already passed through via a default kwarg (`apps=global_apps`)?

Unless I'm misunderstanding your comment?

Reply:
It's defaulting to `apps=global_apps` yes. But if I call it from here and pass the local `apps`, as `create_custom_statuses(app_config, apps=apps, **kwargs)`, then it fails to populate a fresh DB (e.g. the test DB) with errors like:

```
No installed app with label 'circuits'.
```

I'm guessing that this happens because it's being called from `nautobot/extras/migrations/0004_populate_default_status_records.py` and at that early point in the migrations for the `circuits` app haven't necessarily been run yet - but I thought that in migrations we should *always* be using local `apps` rather than `global_apps` and I don't understand why it's working at all with the way it is currently. :-)